backend/amd64/simd.ml

```diff
@@ -443,15 +453,19 @@ let equal_operation_sse41 l r =
   | Multi_sad_u8 l, Multi_sad_u8 r
     when l = r ->
     true
-  | (Round_f64 l, Round_f64 r | Round_f32 l, Round_f32 r) when l = r -> true
+  | Round_sf64 l, Round_sf64 r
+  | Round_f64 l, Round_f64 r
+  | Round_f32 l, Round_f32 r
+    when l = r ->
```

Comment:
(Not really related to this pull request,
but we should probably avoid polymorphic
equality when defining such functions.)